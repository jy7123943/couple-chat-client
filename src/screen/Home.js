import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Image } from 'react-native';
import { commonStyles } from '../styles/Styles';
import { Button, Text } from 'native-base';
import { LinearGradient } from 'expo-linear-gradient';
import * as SecureStore from 'expo-secure-store';

export default function Home (props) {
  const {
    navigation,
    screenProps
  } = props;

  useEffect(() => {
    const authenticateUser = async (navigation, setUserInfo, setRoomInfo) => {
      // await SecureStore.deleteItemAsync('userInfo');
      // await SecureStore.deleteItemAsync('roomInfo');
      const userInfo = await SecureStore.getItemAsync('userInfo');
      const roomInfo = await SecureStore.getItemAsync('roomInfo');
      console.log(userInfo, 'userInfo----------------');
      console.log(roomInfo, 'roomInfo----------------')
      if (userInfo) {
        setUserInfo(JSON.parse(userInfo));

        if (roomInfo) {
          setRoomInfo(JSON.parse(roomInfo));
          return navigation.navigate('Profile');
        }
        return navigation.navigate('CoupleConnect');
      }
      return navigation.navigate('Home');
    };

    if (screenProps.userInfo) {
      return navigation.navigate('CoupleConnect');
    }
    if (screenProps.roomInfo) {
      return navigation.navigate('Profile');
    }

    authenticateUser(navigation, screenProps.setUserInfo, screenProps.setRoomInfo);
  }, []);

  return (
    <LinearGradient
      colors={['#f7dfd3', '#e2c3c8', '#afafc7']}
      style={{
        ...commonStyles.container,
        ...styles.container
      }}
    >
      <View style={styles.viewWrapper}>
        <Image
          source={require('../../assets/couple2.jpg')}
          style={styles.mainImage}
        />
        <Button
          block
          rounded
          style={{
            ...styles.joinButton,
            ...commonStyles.darkBtn
          }}
          onPress={() => {
            navigation.navigate('SignUp');
          }}
        >
          <Text>회원가입</Text>
        </Button>
        <Button
          block
          rounded
          style={commonStyles.lightBtn}
          onPress={() => {
            navigation.navigate('Login');
          }}
        >
          <Text>로그인</Text>
        </Button>
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-end',
    paddingBottom: 100
  },
  joinButton: {
    marginBottom: 10
  },
  viewWrapper: {
    justifyContent: 'flex-end'
  },
  mainImage: {
    width: '100%',
    height: 350,
    borderRadius: 25,
    marginBottom: 80
  }
});
