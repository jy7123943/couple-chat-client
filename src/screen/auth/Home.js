import React, { useEffect } from 'react';
import { StyleSheet, View, Image, Alert } from 'react-native';
import { commonStyles } from '../../styles/Styles';
import { Button, Text } from 'native-base';
import { LinearGradient } from 'expo-linear-gradient';
import * as SecureStore from 'expo-secure-store';

export default function Home (props) {
  const {
    navigation,
    screenProps: {
      setRoomInfo,
      setUserInfo,
      api
    }
  } = props;
  let isMounted = false;

  useEffect(() => {
    isMounted = true;

    const authenticateUser = async (navigation, setUserInfo, setRoomInfo) => {
      try {
        const userInfoString = await SecureStore.getItemAsync('userInfo');

        if (userInfoString) {
          const userInfo = JSON.parse(userInfoString);
          isMounted && setUserInfo(userInfo);

          const response = await api.getUserRoomInfoApi(userInfo.token);

          if (response.result === 'ok') {
            isMounted && setRoomInfo(response.roomInfo);
            return navigation.navigate('Main');
          }

          if (response.result === 'not found') {
            return navigation.navigate('CoupleConnect');
          }
        }
      } catch (err) {
        console.log(err);
        Alert.alert(
          '실패',
          '다시 로그인해주세요',
          [{ text: '확인' }]
        );
        return navigation.navigate('Login');
      }
    };

    if (isMounted) {
      authenticateUser(navigation, setUserInfo, setRoomInfo);
    }

    return () => {
      isMounted = false;
    }
  }, []);

  return (
    <LinearGradient
      colors={['#f7dfd3', '#e2c3c8', '#afafc7']}
      style={{
        ...commonStyles.container,
        ...styles.container
      }}
    >
      <View style={styles.viewWrapper}>
        <Image
          source={require('../../../assets/couple2.jpg')}
          style={styles.mainImage}
        />
        <Button
          block
          rounded
          style={{
            ...styles.joinButton,
            ...commonStyles.darkBtn
          }}
          onPress={() => navigation.navigate('SignUp')}
        >
          <Text>회원가입</Text>
        </Button>
        <Button
          block
          rounded
          style={commonStyles.lightBtn}
          onPress={() => navigation.navigate('Login')}
        >
          <Text>로그인</Text>
        </Button>
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-end',
    paddingBottom: 100
  },
  joinButton: {
    marginBottom: 10
  },
  viewWrapper: {
    justifyContent: 'flex-end'
  },
  mainImage: {
    width: '100%',
    height: 350,
    marginBottom: 80,
    borderRadius: 25
  }
});
