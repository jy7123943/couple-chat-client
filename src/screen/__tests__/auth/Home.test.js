import React from 'react';
import Home from '../../auth/Home';

import { shallow } from 'enzyme';
import { LinearGradient } from 'expo-linear-gradient';
import { Button } from 'native-base';

describe('<Home />', () => {
  let wrapper;
  let navigation;
  let screenProps;

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
    screenProps = {
      setUserInfo: jest.fn(),
      setRoomInfo: jest.fn()
    };
    wrapper = shallow(
      <Home
        navigation={navigation}
        screenProps={screenProps}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly', () => {
    expect(wrapper.find(LinearGradient)).toHaveLength(1);
    expect(wrapper.find(Button)).toHaveLength(2);
  });

  it('navigate to SignUp if user press sign up button', () => {
    wrapper.find(Button).first().simulate('press');
    expect(navigation.navigate).toHaveBeenCalledWith('SignUp');
  });

  it('navigate to Login if user press login button', () => {
    wrapper.find(Button).last().simulate('press');
    expect(navigation.navigate).toHaveBeenCalledWith('Login');
  });
});
