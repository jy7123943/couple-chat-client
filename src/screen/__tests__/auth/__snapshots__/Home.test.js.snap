// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Home screen snapshot 1`] = `ShallowWrapper {}`;
